### PR TITLE
fix: use the new slug fields on the frontend side

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- Queries the new slug and uses it for modal navigation
- Everything else works auto-magically due to fixing commentPermalink attribute
- Related API changes: https://github.com/dailydotdev/daily-api/pull/1801

Open question:
Edit 2: Canonical working
Screenshot 2024-04-04 at 16 49 09

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

AS-227 #done

### CODE REVIEW

File: packages/webapp/pages/posts/[id]/index.tsx [hunk 1]
```diff
@@ -99,6 +99,7 @@ const PostPage = ({ id, initialData }: Props): ReactElement => {
     return post?.title;
   };
   const seo: NextSeoProps = {
+    canonical: post?.slug,
```
Comment (idoshamun) [inline packages/webapp/pages/posts/[id]/index.tsx:102]: It's not enough for the canonical isn't it? You miss the `/posts/` part

Comment (rebelchris) [reply packages/webapp/pages/posts/[id]/index.tsx:102]: You're absolutely right 😶‍🌫️

File: packages/webapp/pages/posts/[id]/index.tsx [hunk 2]
```diff
@@ -99,6 +100,7 @@ const PostPage = ({ id, initialData }: Props): ReactElement => {
     return post?.title;
   };
   const seo: NextSeoProps = {
+    canonical: `${webappUrl}/posts/${post?.slug}`,
```
Comment (capJavert) [inline packages/webapp/pages/posts/[id]/index.tsx:103]:
```suggestion
    canonical: post?.slug ? `${webappUrl}/posts/${post.slug}` : undefined,
```

Comment (capJavert) [reply packages/webapp/pages/posts/[id]/index.tsx:103]: Just in case of `fallback`

Comment (rebelchris) [reply packages/webapp/pages/posts/[id]/index.tsx:103]: Ah good point yes!

**Review by idoshamun — APPROVED** [review]

**Review by capJavert — APPROVED** [review]

**Pull request merged**